src/ToDoList.js

```diff
@@ -18,177 +16,142 @@ import {
-  backgroundColor: "lightGrey",
-  borderRadius: "30px",
-  margin: "0 auto",
-  marginTop: "30px",
-};
-
-const titleStyle = {
-  color: "black",
-  variant: "h1",
-  width: "300px",
-  height: "50px",
-  //borderRadius: "20px",
-  //backgroundColor: "lightBlue",
-  margin: "0 auto",
-  fontWeight: "bold",
-  //justifyContent: "center",
-  //alignItems: "center",
-};
-
-const inputContainerStyle = {
-  display: "flex",
-  justifyContent: "center",
-  alignItems: "center",
-  //height: "40px",
-};
-
-/*const taskItemStyle = {
-  width: "200px",
-  display: "flex",
-  display: "flex",
-  justifyContent: "center",
-  alignItems: "center",
-};*/
-
-const completedTaskStyle = {
-  textDecoration: "line-through",
-  textDecorationStyle: "solid",
-  textDecorationThickness: "2px",
-  color: "black",
-};
-
-/*const deleteButtonStyle = {
-  fontSize: "10px",
-  padding: "3px 3px",
-};*/
+const containerStyle = css`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  gap: 30px;
+  h1 {
+    color: #29b6f6;
+    font-size: 50px;
+    font-weight: bold;
+    margin-top: 30px;
+  }
+`;
+
+const completedTaskStyle = css`
+  text-decoration: line-through solid 2px;
+  color: black;
+`;
+
+const listContainerStyle = css`
+  width: 100%;
+  max-height: 450px;
+  overflow-x: hidden;
+  overflow-y: auto;
+`;
 
 const ToDoList = () => {
-  useEffect(() => {
-    const storedTasks = localStorage.getItem("tasks");
-    if (storedTasks) {
-      setTasks(JSON.parse(storedTasks));
-    }
-  }, []);
-
-  const [tasks, setTasks] = useState(() => {
-    const storedTasks = localStorage.getItem("tasks");
+  const [newTaskValue, setNewTaskValue] = useState(() => {
+    const storedTasks = localStorage.getItem("newTaskValue");
     return storedTasks ? JSON.parse(storedTasks) : [];
   });
 
-  const [taskInfo, setTaskInfo] = useState("");
-  const [filter, setFilter] = useState("all");
-
-  useEffect(() => {
-    localStorage.setItem("tasks", JSON.stringify(tasks));
-  }, [tasks]);
+  const [inputValue, setInputValue] = useState("");
+  const [taskState, associateTaskState] = useState("all");
 
-  const handleInputChange = (event) => {
-    setTaskInfo(event.target.value);
-  };
   const addTask = () => {
-    if (taskInfo.trim() !== "") {
-      const newTask = {
-        id: Date.now(),
-        task: taskInfo,
-        completed: false,
-      };
-
-      setTasks([...tasks, newTask]);
-      setTaskInfo("");
+    if (inputValue.trim() === "") {
+      alert("Trying to add an empty string!");
+      return;
     }
-  };
 
-  const deleteTask = (taskId) => {
-    const updatedTasks = tasks.filter((task) => task.id !== taskId);
-    setTasks(updatedTasks);
+    const isTaskExists = newTaskValue.some(
+      (task) => task.task === inputValue.trim()
+    );
+
+    if (isTaskExists) {
+      alert("Task already exists!");
+      return;
+    }
+
+    const newTask = {
+      id: Date.now(),
+      task: inputValue.trim(),
```

Comment:
maybe `description` will be a better name ?
because I saw that you're accessing this property later like this: `task.task`, which is confusing.

Reply:
Done.